broker/src/main/java/io/zeebe/broker/system/monitoring/DiskSpaceUsageMonitor.java

```diff
@@ -56,11 +57,18 @@ private void checkDiskUsageAndNotifyListeners() {
   }
 
   public void addDiskUsageListener(final DiskSpaceUsageListener listener) {
-    diskSpaceUsageListeners.add(listener);
+    actor.call(
+        () -> {
+          diskSpaceUsageListeners.add(listener);
+          // Listeners always assumes diskspace is available on start. So notify them if it is not.
+          if (!currentDiskAvailableStatus) {
+            listener.onDiskSpaceNotAvailable();
+          }
+        });
   }
 
   public void removeDiskUsageListener(final DiskSpaceUsageListener listener) {
-    diskSpaceUsageListeners.remove(listener);
+    actor.call(() -> diskSpaceUsageListeners.remove(listener));
```

Comment:
Why this change? Why no longer do it immediately?

Reply:
This will be invoked by other actors. So to prevent concurrent access to `diskSpaceUsageListeners` we submit the task to this actor so that only this actor can modify it.